asf_smart_meter_exploration/analysis/clustering.py

```diff
@@ -0,0 +1,129 @@
+# File: asf_smart_meter_exploration/analysis/clustering.py
+"""
+Script to perform clustering on variants of the smart meter data.
+For each cluster, plots of the counts and the distribution of
+tariffs / Acorn groups in each cluster are produced and saved.
+"""
+
+from asf_smart_meter_exploration.utils.clustering_utils import run_clustering
+from asf_smart_meter_exploration.utils.plotting_utils import *
+from asf_smart_meter_exploration.pipeline.data_aggregation import *
+
+# Dictionary of variants to cluster and plot.
+# "df" is the smart meter dataframe, "k" is the number of clusters,
+# and see docs for `plot_observations_and_clusters` (in `utils/plotting_utils.py`)
+# for other parameters.
+# Values of k here were chosen after analysing plots produced in
+# `analysis/inertia_plots.py`.
+options_dict = {
```

Comment:
Not sure whether this dictionary should be in a config file or here (according to [this](https://docs.google.com/presentation/d/1F2vYPstP2sRAysvmnK0wHpCBTl-Vou_m9jraN7vX4E0/edit#slide=id.g175b4677f73_0_233))

I think maybe a config file? Otherwise the `inertia_plots.py` becomes dependent on this, and all modules in analysis should be independent of each other as per the slide deck above.

Reply:
Agree, keeping it in a config file is better!